app/src/main/java/de/tum/in/tumcampusapp/managers/CalendarManager.java

```diff
@@ -219,10 +188,20 @@ public void deleteLectureFromBlacklist(int widgetId, String lecture) {
      * @param widgetId the Id of the widget
      * @return A cursor containing a list of lectures and the is_on_blacklist flag
      */
-    public Cursor getLecturesFromWidget(int widgetId) {
-        return db.rawQuery("SELECT DISTINCT c.ROWID as _id, c.title, EXISTS (" +
-                           "SELECT * FROM widgets_timetable_blacklist WHERE widget_id=? AND lecture_title=c.title" +
-                           ") as is_on_blacklist from calendar c GROUP BY c.title", new String[]{String.valueOf(widgetId)});
+    public List<CalendarItem> getLecturesForWidget(int widgetId) {
+        List<CalendarItem> lectures = new ArrayList<>();
+        List<CalendarItem> distinctLectures = calendarDao.getDistinctLectures();
+        List<CalendarItem> blacklistedLectures = calendarDao.getLecturesWithBlacklist(Integer.toString(widgetId));
+        for (CalendarItem distinctLecture: distinctLectures) {
+            for (CalendarItem blacklistedLecture: blacklistedLectures) {
+                if (distinctLecture.getTitle().equals(blacklistedLecture.getTitle())) {
```

Comment:
This logic looks like a manual `JOIN` to me... Is there a good reason doing this in Java vs. just SQL?

Reply:
This would require an additional entity just for that specific use case, I decided to re-use calendar-item with non-persisted field, since the blacklist contains only blacklisted items.

Reply:
With a second thought:
Can't you just query all non-blacklisted lectures (inverse of `getLecturesWithBlacklist()`) and merge those two lists?
I.e.
```java
List<CalendarItem> lectures = getBlacklistedLectures(widgetId);
for (CalendarItem i : lectures) {
    i.setBlacklisted(true);
}
lectures.addAll(getNonBlacklistedLectures(widgetId))
```

I really dislike the idea performing an O(n²) operation for a potentially large amount of lectures...

Reply:
That should work!